#VRML_SIM R2021a utf8
WorldInfo {
  coordinateSystem "NUE"
  contactProperties [
    ContactProperties {
      material1 "m1"
      material2 "m2"
      coulombFriction [
        0
      ]
      bounce 0
      bounceVelocity 0
    }
  ]
}
Viewpoint {
  orientation -0.8461672388667071 -0.48798129294081694 -0.21418511061346437 0.9847141687023749
  position -0.30002016052900954 0.35073157197007887 0.5215644633174278
}
TexturedBackground {
}
TexturedBackgroundLight {
}
RectangleArena {
  translation -0.00776461 0 -0.0389779
  rotation 0 1 0 0.261799
  contactMaterial "m2"
  floorSize 2 1
  floorTileSize 2 1
  floorAppearance PBRAppearance {
    baseColorMap ImageTexture {
      url [
        "Arena_LineFollower.png"
      ]
    }
    metalness 0
  }
}
Robot {
  translation -0.14192800675859468 0.0006061442684835508 0.2589589747767459
  rotation 1.3031044416347678e-05 0.9999999997708946 -1.6982416374027285e-05 -1.3089953074008833
  children [
    DistanceSensor {
      translation -0.03 0.02 0.042
      rotation 0 0 1 -1.57
      children [
        Shape {
          appearance PBRAppearance {
            roughness 1
            metalness 0
          }
          geometry Box {
            size 0.005 0.005 0.005
          }
        }
      ]
      name "ir2"
      type "infra-red"
      numberOfRays 10
      aperture 0.1
    }
    DistanceSensor {
      translation 0.03 0.02 0.042
      rotation 0 0 -1 1.57
      children [
        Shape {
          appearance PBRAppearance {
            roughness 1
            metalness 0
          }
          geometry Box {
            size 0.005 0.005 0.005
          }
        }
      ]
      name "ir1"
      type "infra-red"
      numberOfRays 10
      aperture 0.1
    }
    DistanceSensor {
      translation 0 0.02 0.05
      rotation 0 0 1 -1.57
      children [
        Shape {
          appearance PBRAppearance {
            roughness 1
            metalness 0
          }
          geometry Box {
            size 0.005 0.005 0.005
          }
        }
      ]
      name "ir0"
      type "infra-red"
      numberOfRays 10
      aperture 0.1
    }
    DEF body Transform {
      translation 0 0.05 0
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0.7 0.5 1
            roughness 1
            metalness 0
          }
          geometry Cylinder {
            height 0.1
            radius 0.05
          }
        }
      ]
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        position -4.8630524088517507e-17
        anchor 0.05 0.039 0
      }
      device [
        RotationalMotor {
          name "motor2"
        }
      ]
      endPoint Solid {
        translation 0.05 0.039 0
        rotation -1.7034242856848286e-16 1.702068343098113e-16 1 1.57
        children [
          DEF wheel Shape {
            appearance PBRAppearance {
              roughness 1
              metalness 0
            }
            geometry Cylinder {
              height 0.02
              radius 0.04
            }
          }
        ]
        name "solid(1)"
        boundingObject USE wheel
        physics Physics {
        }
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        position 1.236460169108222e-16
        anchor -0.05 0.039 0
      }
      device [
        RotationalMotor {
          name "motor1"
        }
      ]
      endPoint Solid {
        translation -0.050008 0.039 0
        rotation 5.901776142610404e-16 -4.327610591878777e-16 1 1.57
        children [
          DEF wheel Shape {
            appearance PBRAppearance {
              roughness 1
              metalness 0
            }
            geometry Cylinder {
              height 0.02
              radius 0.04
            }
          }
        ]
        boundingObject USE wheel
        physics Physics {
        }
      }
    }
  ]
  contactMaterial "m1"
  boundingObject USE body
  physics Physics {
  }
  controller "robotics_round_2"
}
